tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_CcsnCollapse.cpp

```diff
@@ -0,0 +1,250 @@
+      Catch::Matchers::Contains("Data file not found"));
+
+  const auto ccsn_progenitor =
+      serialize_and_deserialize(ccsn_progenitor_original);
+  CHECK(ccsn_progenitor == CcsnCollapse(progenitor_filename,
+                                        polytropic_constant, adiabatic_index,
+                                        central_angular_velocity,
+                                        diff_rot_parameter, max_dens_ratio));
+  CHECK(ccsn_progenitor !=
+        CcsnCollapse(progenitor_filename, polytropic_constant + 0.1,
+                     adiabatic_index, central_angular_velocity,
+                     diff_rot_parameter, max_dens_ratio));
+  CHECK(ccsn_progenitor !=
+        CcsnCollapse(progenitor_filename, polytropic_constant,
+                     adiabatic_index + 0.1, central_angular_velocity,
+                     diff_rot_parameter, max_dens_ratio));
+  CHECK(ccsn_progenitor != CcsnCollapse(progenitor_filename,
+                                        polytropic_constant, adiabatic_index,
+                                        central_angular_velocity + 0.1,
+                                        diff_rot_parameter, max_dens_ratio));
+  CHECK(ccsn_progenitor !=
+        CcsnCollapse(progenitor_filename, polytropic_constant, adiabatic_index,
+                     central_angular_velocity, diff_rot_parameter + 100,
+                     max_dens_ratio));
+}
+
+void test_ccsn_collapse(std::string progenitor_filename,
+                        double polytropic_constant, double adiabatic_index,
+                        double central_angular_velocity,
+                        double diff_rot_parameter, double max_dens_ratio) {
+  Parallel::register_classes_with_charm<grmhd::AnalyticData::CcsnCollapse>();
+
+  const std::unique_ptr<evolution::initial_data::InitialData> option_solution =
+      TestHelpers::test_option_tag_factory_creation<
+          evolution::initial_data::OptionTags::InitialData,
+          grmhd::AnalyticData::CcsnCollapse>(
+          "CcsnCollapse:\n"
+          "  ProgenitorFilename: " +
+          progenitor_filename +
+          "\n"
+          "  AdiabaticIndex: " +
+          std::to_string(adiabatic_index) +
+          "\n"
+          "  CentralAngularVelocity: " +
+          std::to_string(central_angular_velocity) +
+          "\n"
+          "  DifferentialRotationParameter: " +
+          std::to_string(diff_rot_parameter) +
+          "\n"
+          "  PolytropicConstant: " +
+          std::to_string(polytropic_constant) +
+          "\n"
+          "  MaxDensityRatioForLinearInterpolation: " +
+          std::to_string(max_dens_ratio) + "\n")
+          ->get_clone();
+
+  const auto deserialized_option_solution =
+      serialize_and_deserialize(option_solution);
+  const auto& ccsn_progenitor =
+      dynamic_cast<const grmhd::AnalyticData::CcsnCollapse&>(
+          *deserialized_option_solution);
+
+#ifdef SPECTRE_DEBUG
+  const std::unique_ptr<evolution::initial_data::InitialData>
```

Comment:
I'm missing something in terms of the code inside this `ifdef`. I don't see a `CHECK` or `CHECK_THROWS_WITH`. Is one of those supposed to be there? Oh! It's used below on 188. Could you move the create, etc. down there so it's clearer how it's used?